cachi2/core/package_managers/npm.py

```diff
@@ -173,6 +185,117 @@ def get_sbom_components(self) -> list[dict[str, str]]:
         packages = self._dependencies if self.lockfile_version == 1 else self._packages
         return [{"name": package.name, "version": package.version} for package in packages]
 
+    def get_dependencies_to_download(self) -> Dict[str, Dict[str, Any]]:
+        """Return a list of URL dependencies to download."""
+        packages = self._dependencies if self.lockfile_version == 1 else self._packages
+        return {
+            package.name: {
+                "version": package.version,
+                "url": package.resolved_url,
+                "integrity": package.integrity,
+            }
+            for package in packages
+            if "file:" not in package.resolved_url
+        }
+
+
+def _get_external_requirement_filepath(
+    download_path: RootedPath, package: str, requirement_url: str
+) -> Path:
+    if requirement_url.startswith("https://"):
+        algorithm = "sha512"
+        digest = "YOLO"
+        # TODO: Do I have to download that package first to get algorithm and digest???
```

Comment:
Question:  Do I have to download that package first to get algorithm and digest?

Reply:
it's not typical checksum as I mentioned in other comment, it's base64 encoded one but for our purposes should be sufficient.

Reply:
You can convert it to the representation you want https://github.com/containerbuildsystem/cachito/blob/db612e6022311644913c611e3553097490cc94b1/cachito/workers/pkg_managers/general_js.py#L780